src/instrumentations/extentions/aws/SQSEventCreator.ts

```diff
@@ -0,0 +1,128 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Span } from '@opentelemetry/api';
+import {
+  AwsSdkRequestHookInformation,
+  AwsSdkResponseHookInformation,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWSEventCreator } from './event-creator-interface';
+
+export class SQSEventCreator implements AWSEventCreator {
+  requestHandler(span: Span, requestInfo: AwsSdkRequestHookInformation): void {
+    const queueUrl = requestInfo.request.commandInput.QueueUrl;
+    const splited = queueUrl.split('/');
+    span.setAttribute('aws.sqs.queue_name', splited.pop());
+    span.setAttribute('aws.account_id', splited.pop());
+    // span.setAttribute('aws.region', `${requestInfo.request.region}`);
+
+    switch (requestInfo.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          'aws.sqs.record.message_body',
+          requestInfo.request.commandInput.MessageBody
+        );
+        // span.setAttribute(
+        //   'aws.sqs.record.message_attributes',
+        //   JSON.stringify(requestInfo.request.commandInput.MessageAttributes)
+        // );
+        span.setAttribute(
+          'aws.sqs.record.delay_seconds',
+          requestInfo.request.commandInput.DelaySeconds
+        );
+        break;
+      case 'SendMessageBatch':
+        span.setAttribute(
+          'aws.sqs.entries',
+          JSON.stringify(requestInfo.request.commandInput.Entries)
+        );
+        break;
+      case 'ReceiveMessage':
+        // span.setAttribute(
+        //   'aws.sqs.record.message_attributes',
+        //   requestInfo.request.commandInput.MessageAttributeNames
+        // );
+        span.setAttribute(
+          'aws.sqs.visibility_timeout',
+          requestInfo.request.commandInput.VisibilityTimeout
+        );
+        span.setAttribute(
+          'aws.sqs.wait_time_seconds',
+          requestInfo.request.commandInput.WaitTimeSeconds
+        );
+        span.setAttribute(
+          'aws.sqs.max_number_of_messages',
+          requestInfo.request.commandInput.MaxNumberOfMessages
+        );
+        break;
+    }
+  }
+  responseHandler(
+    span: Span,
+    responseInfo: AwsSdkResponseHookInformation
+  ): void {
+    switch (responseInfo.response.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          'aws.sqs.record.message_id',
+          responseInfo.response.data.MessageId
+        );
+        break;
+      case 'SendMessageBatch':
+        span.setAttribute(
+          'aws.sqs.record',
+          JSON.stringify(responseInfo.response.data)
+        );
+        break;
+      case 'ReceiveMessage': {
+        if (responseInfo.response.data.Messages.length == 1) {
+          span.setAttribute(
+            'aws.sqs.record.message_body',
+            responseInfo.response.data.Messages[0].Body
```

Comment:
same https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md.

The attributes labels should be flat